### ISSUE TITLE
Improve example.html files

### ISSUE THREAD

**Issue by jasomdotnet** [ghybs/Leaflet.MarkerCluster.LayerSupport#17]
Example.html files are terrible, impossible to reproduce with own configuration. Where is configuration code (map definition, for example)? There is no plain code in source.

```
mcgLayerSupport-controlLayers-realworld.100k.html
mcgLayerSupport-controlLayers-realworld.388.html
etc..
```

I'm forced to reverse-engineering the code (removing line after the line). It reminds me times of Internet explorer (Yes, so bad examples are).

**Comment by jasomdotnet** [comment]
OMFG, I found it, what a feeling! It is in mcgLayerSupport-controlLayers-realworld.388.js file in the same folder. #ReverseEngineeringRetro

**Comment by jasomdotnet** [comment]
Okey, here we go. This is first locally working example:

```
<!DOCTYPE html>
<html lang="en">
    <head>
	<meta charset="utf-8" />
	<title>Example</title>
	<meta name="viewport" content="width=device-width, initial-scale=1.0">
	<link rel="stylesheet" href="screen.css" />
	<script src="https://leaflet.github.io/Leaflet.markercluster/example/realworld.50000.1.js"></script>
	<script src="https://leaflet.github.io/Leaflet.markercluster/example/realworld.50000.2.js"></script>
    </head>
    <body>
	<div id="progress"><div id="progress-bar"></div></div>
	<div id="map"></div>

	<link rel="stylesheet" href="leaflet.css" />
	<script src="leaflet-src.js"></script>
	<link rel="stylesheet" href="MarkerCluster.css" />
	<link rel="stylesheet" href="MarkerCluster.Default.css" />
	<script src="leaflet.markercluster-src.js"></script>
	<script src="leaflet.markercluster.layersupport-src.js"></script>
	<script >
	    var tiles = L.tileLayer( 'https://{s}.tile.openstreetmap.org/{z}/{x}/{y}.png', {
		maxZoom: 18,
		attribution: '&copy; <a href="https://www.openstreetmap.org/copyright">OpenStreetMap</a> contributors, Points &copy 2012 LINZ'
	    } ),
		latlng = L.latLng( -37.82, 175.24 );


	    var map = L.map( 'map', { center: latlng, zoom: 13, layers: [ tiles ] } );

	    var progress = document.getElementById( 'progress' );
	    var progressBar = document.getElementById( 'progress-bar' );

	    var mcgLayerSupportGroup = L.markerClusterGroup.layerSupport( {
		chunkedLoading: true,
		chunkInterval: 100,
		chunkProgress: updateProgressBar
	    } ),
		control = L.control.layers( null, null, { collapsed: false } );

	    var group1 = L.layerGroup( fillMarkerList( addressPoints ) ); // 25k points
	    var group2 = L.layerGroup( fillMarkerList( addressPoints2 ) ); // 25k points
	    var group3 = L.layerGroup( fillMarkerList( addressPoints ) ); // 25k points
	    var group4 = L.layerGroup( fillMarkerList( addressPoints2 ) ); // 25k points
	    //var superGroup = L.layerGroup([group1, group2, group3, group4]);

	    mcgLayerSupportGroup.addTo( map );

	    mcgLayerSupportGroup.checkIn( [ group1, group2, group3, group4 ] );

	    control.addOverlay( group1, 'First quarter' );
	    control.addOverlay( group2, 'Second quarter' );
	    control.addOverlay( group3, 'Third quarter' );
	    control.addOverlay( group4, 'Fourth quarter' );
	    //control.addOverlay(superGroup, 'Super Group');
	    control.addTo( map );

	    group1.addTo( map ); // Adding to map or to AutoMCG are now equivalent.
	    group2.addTo( map );
	    group3.addTo( map );
	    group4.addTo( map );
	    //mcgLayerSupportGroup.addLayer(superGroup);

	    function updateProgressBar( processed, total, elapsed, layersArray ) {
		console.log( "updateProgressBar" );
		if ( elapsed > 200 ) {
		    // if it takes more than a second to load, display the progress bar:
		    progress.style.display = 'block';
		    progressBar.style.width = Math.round( processed / total * 100 ) + '%';
		}
		if ( processed === total ) {
		    // all markers processed - hide the progress bar:
		    progress.style.display = 'none';
		}
	    }

	    function fillMarkerList( addressPointsSource ) {
		var markerList = [ ];
		for ( var i = 0; i < addressPointsSource.length; i++ ) {
		    var a = addressPointsSource[i];
		    var title = a[2];
		    var marker = L.marker( L.latLng( a[0], a[1] ), {
			title: title
		    } );
		    marker.bindPopup( title );
		    markerList.push( marker );
		}
		return markerList;
	    }

	</script>
    </body>
</html>
```

Pasting the code here so it may help somebody. Also linked this template from [here](https://stackoverflow.com/questions/40482919/cluster-markers-from-different-overlays-with-each-other#comment104426870_40547694).

Note: do not use resources left on unpkg.com, download them locally. You will find download links in example files when you open console.

**Comment by ghybs** [comment]
Hi,

I am glad you managed to get an example to your liking.

It is true the demo was not meant to be reproduced, but only to provide a live trial.

The [quick guide](https://github.com/ghybs/Leaflet.MarkerCluster.LayerSupport/blob/master/README.md#quick-guide) was there for that. But maybe a working bootstrapping example would have provided an intermediate step.

**Comment by jasomdotnet** [comment]
This morning I glued Leaflet.MarkerCluster.LayerSupport in my app and it is perfect!

I saw that quick guide text and for me it was too technical (js guy writes for js guys).

I was disappointed with "live trial" source code, because there was no direct script configuration.

For example [original Leafleat tutorials](https://leafletjs.com/examples/quick-start/example.html) are easy to follow. Plug-ins should follow the pattern.
